firmware/panic-probe/src/lib.rs

```diff
@@ -64,26 +67,22 @@ mod imp {
 /// ```
 /// #[defmt::panic_handler]
 /// fn panic() -> ! {
-///     panic_probe::hard_fault();
+///     panic_probe::abort();
 /// }
 /// ```
 #[cfg(target_os = "none")]
-pub fn hard_fault() -> ! {
-    // If `UsageFault` is enabled, we disable that first, since otherwise `udf` will cause that
-    // exception instead of `HardFault`.
-    #[cfg(not(any(armv6m, armv8m_base)))]
-    {
-        const SHCSR: *mut u32 = 0xE000ED24usize as _;
-        const USGFAULTENA: usize = 18;
-
-        unsafe {
-            let mut shcsr = core::ptr::read_volatile(SHCSR);
-            shcsr &= !(1 << USGFAULTENA);
-            core::ptr::write_volatile(SHCSR, shcsr);
-        }
-    }
+pub fn abort() -> ! {
+    semihosting::process::abort();
+}
 
-    cortex_m::asm::udf();
+#[cfg(target_os = "none")]
+pub fn disable_isr() {
```

Comment:
I think I would prefer `fn cortex_m_disable_isr()` and `fn riscv32_disable_isr()` and then use the `[cfg]` to `use crate::cortex_m_disable_isr as disable_isr;`.

Then if the neither cfg is `true`, you get a missing function error rather than a silent miscompilation that doesn't disable interrupts.

Also, I wonder if we should be using the critical-section crate, because disabling interrupts is insufficient on a multi-core system.